src/Raven.Server/Monitoring/OpenTelemetry/MetricsManager.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Raven.Client.Util;
+using Raven.Server.Config;
+using Raven.Server.ServerWide.Context;
+
+namespace Raven.Server.Monitoring.OpenTelemetry;
+
+public class MetricsManager
+{
+    private readonly RavenServer _server;
+    private readonly SemaphoreSlim _locker = new(1, 1);
+    private ServerMetrics _serverMetrics;
+    
+    public MetricsManager(RavenServer server)
+    {
+        _server = server;
+        _server.ServerStore.LicenseManager.LicenseChanged += OnLicenseChanged;
+    }
+
+    private void OnLicenseChanged()
+    {
+        if (_server.Configuration.Monitoring.OpenTelemetry.Enabled == false)
+            return;
+
+        _locker.Wait();
+        try
+        {
+            var activate = _server.ServerStore.LicenseManager.CanUseOpenTelemetryMonitoring(withNotification: true, startUp: false);
+            if (activate)
```

Comment:
you are not taking into account here that you might already ran it, this even can fire few times during single run, in SNMP we are checking the state and if it was already active then we are not doing anything

Reply:
Added a boolean value as an indicator to show whether we registered metrics (and skip if we already did it).